lib/bulk_insert/worker.rb

```diff
@@ -5,27 +5,30 @@ class Worker
     attr_accessor :before_save_callback
     attr_accessor :after_save_callback
     attr_accessor :adapter_name
-    attr_reader :ignore, :update_duplicates
+    attr_reader :ignore, :update_duplicates, :result_sets
 
-    def initialize(connection, table_name, column_names, set_size=500, ignore=false, update_duplicates=false)
+    def initialize(connection, table_name, primary_key, column_names, set_size=500, ignore=false, update_duplicates=false, return_primary_keys=false)
       @connection = connection
       @set_size = set_size
 
       @adapter_name = connection.adapter_name
       # INSERT IGNORE only fails inserts with duplicate keys or unallowed nulls not the whole set of inserts
       @ignore = ignore
       @update_duplicates = update_duplicates
+      @return_primary_keys = return_primary_keys
```

Comment:
nb: I think `return_primary_keys` is redundant. It looks like `return_primary_keys = true` is implied by passing in an argument for `primary_key`.

Reply:
The user never passes primary_key. primary_key is retrieved from the active record class. So it's always there.